modules/tinymce/src/core/main/ts/api/dom/DOMUtils.ts

```diff
@@ -1090,43 +1089,45 @@ function DOMUtils(doc: Document, settings: Partial<DOMUtilsSettings> = {}): DOMU
     }
   };
 
-  const bind = <T = any>(target: Target, name: string, func: EventUtilsCallback<T>, scope?: any): EventUtilsCallback<T>[] | EventUtilsCallback<T> => {
-    if (Tools.isArray(target)) {
+  const bind = <
+    K extends string,
+    T extends Target,
+    F extends EventUtilsCallback<MappedEvent<HTMLElementEventMap, K>>
+  >(target: T, name: string, func: F, scope?: any): T extends [] ? F[] : F => {
+    if (Tools.isArray<Node | Window>(target)) {
       let i = target.length;
-      const rv = [];
+      const rv = [] as F[];
```

Comment:
Please don't use casts, this should be `const rv:  F[] = []`. Also `F` is quite uncommon, why'd you use that? Our normal convention is to use `U` here.

Reply:
F was just because it was a function, I can change it to U if that's more regular. Will remove this cast — not sure if there's anything I can do about the casts below that will still compile